lib/queue.js

```diff
@@ -2,6 +2,63 @@
 
 module.exports = (api) => {
 
+  api.metasync.queue = (concurrency) => {
+    let active = true;
+    let count = 0;
+    let wait = 0;
+    let timeout = 0;
+    const input = [];
+    const throttles = [];
+    let process = api.common.emptiness;
+    let output = api.common.emptiness;
+    let drain = api.common.emptiness;
+
+    const queue = (item) => {
+      if (active && count < concurrency) queue.next(item);
+      else input.push(item);
+      let timer;
+      if (wait) {
+        timer = setTimeout(() => {
+          output(new Error('Queue wait timeout'));
```

Comment:
These lines do not have such a goal. We use `queue.next(item)` if queue have empty slots and `imput.push(item)` if it haven't